domain/status/state/state.go

```diff
@@ -1162,3 +1165,269 @@ ON CONFLICT(unit_uuid) DO UPDATE SET
 	}
 	return nil
 }
+
+// GetApplicationAndUnitStatuses returns the application and unit statuses of
+// all the applications in the model, indexed by application name.
+func (st *State) GetApplicationAndUnitStatuses(ctx context.Context) (map[string]status.Application, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	// Get all the applications.
+	applicationQuery, err := st.Prepare(`
+SELECT a.name AS &applicationStatusDetails.name,
+	a.uuid AS &applicationStatusDetails.uuid,
+	a.life_id AS &applicationStatusDetails.life_id,
+	ap.os_id AS &applicationStatusDetails.platform_os_id,
+	ap.channel AS &applicationStatusDetails.platform_channel,
+	ap.architecture_id AS &applicationStatusDetails.platform_architecture_id,
+	ac.track AS &applicationStatusDetails.channel_track,
+	ac.risk AS &applicationStatusDetails.channel_risk,
+	ac.branch AS &applicationStatusDetails.channel_branch,
+	cm.subordinate AS &applicationStatusDetails.subordinate,
+	s.status_id AS &applicationStatusDetails.status_id,
+	s.message AS &applicationStatusDetails.message,
+	s.data AS &applicationStatusDetails.data,
+	s.updated_at AS &applicationStatusDetails.updated_at,
+	re.relation_uuid AS &applicationStatusDetails.relation_uuid,
+	c.reference_name AS &applicationStatusDetails.charm_reference_name,
+	c.revision AS &applicationStatusDetails.charm_revision,
+	c.source_id AS &applicationStatusDetails.charm_source_id,
+	c.architecture_id AS &applicationStatusDetails.charm_architecture_id,
+	c.version AS &applicationStatusDetails.charm_version,
+	c.lxd_profile AS &applicationStatusDetails.lxd_profile,
+	aps.scale AS &applicationStatusDetails.scale,
+	k8s.provider_id AS &applicationStatusDetails.k8s_provider_id,
+	EXISTS(SELECT 1 FROM v_application_exposed_endpoint AS ae WHERE ae.application_uuid = a.uuid) AS &applicationStatusDetails.exposed
+FROM application AS a
+JOIN application_platform AS ap ON ap.application_uuid = a.uuid
+LEFT JOIN application_channel AS ac ON ac.application_uuid = a.uuid
+JOIN charm AS c ON c.uuid = a.charm_uuid
+JOIN charm_metadata AS cm ON cm.charm_uuid = c.uuid
+LEFT JOIN application_status AS s ON s.application_uuid = a.uuid
+LEFT JOIN k8s_service AS k8s ON k8s.application_uuid = a.uuid
+LEFT JOIN application_scale AS aps ON aps.application_uuid = a.uuid
+LEFT JOIN v_relation_endpoint AS re ON re.application_uuid = a.uuid
+ORDER BY a.name, re.relation_uuid;
```

Comment:
On a previous PR of mine, @wallyworld asked for alignment on the longest SQL keyword(s), in this case it would be on `ORDER BY`. I don't have a strong opinion on whether we should or shouldn't do it, but we should be consistent.

Reply:
Perhaps a view?